### PR TITLE
Stresser v2

### PR DESCRIPTION
**Pull request by chaochn47**
*Description of changes:*

Added stresser v2 add on to keep updating one k8s resource until reaching etcd limit. 

```bash
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_ENABLE=true \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_REPOSITORY_ACCOUNT_ID=730249423355 \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_REPOSITORY_NAME=aws/aws-k8s-tester \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_REPOSITORY_IMAGE_TAG=latest \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_REPOSITORY_BUSYBOX_NAME=busybox \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_REPOSITORY_BUSYBOX_IMAGE_TAG=latest \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SCHEDULE='0 */6 * * *' \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_PARALLELS=10 \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_COMPLETES=10 \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_OBJECT_SIZE=8 \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_DURATION=15m \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_COROUTINE=10 \
AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SECRET_NUM=10 \
aws-k8s-tester eks create cluster --enable-prompt=true --auto-path
```

```bash
ESC[33m*********************************
ESC[0mESC[92mtesters[34].Create ESC[36m"github.com/aws/aws-k8s-tester/eks/stresser2" ESC[39m("/tmp/eks-2021022507-spheres8fvv3.yaml", "/tmp/kubectl-test-v1.18.9 --kubeconfig=/tmp/eks-2021022507-spheres8fvv3.kubeconfig.yaml")
ESC[0m{"level":"info","ts":"2021-02-25T08:46:47.473Z","caller":"stresser2/stresser.go:72","msg":"starting tester.Create","tester":"github.com/aws/aws-k8s-tester/eks/stresser2"}
{"level":"info","ts":"2021-02-25T08:46:47.487Z","caller":"ecr/ecr.go:39","msg":"describing an ECR repository","repo-account-id":"730249423355","repo-region":"us-west-2","repo-name":"aws/aws-k8s-tester","image-tag":"latest","image":"730249423355.dkr.ecr.us-west-2.amazona
ws.com/aws/aws-k8s-tester:latest"}
{"level":"info","ts":"2021-02-25T08:46:47.510Z","caller":"ecr/ecr.go:72","msg":"described an ECR repository","repo-arn":"arn:aws:ecr:us-west-2:730249423355:repository/aws/aws-k8s-tester","repo-region":"us-west-2","repo-name":"aws/aws-k8s-tester","repo-uri":"730249423355
.dkr.ecr.us-west-2.amazonaws.com/aws/aws-k8s-tester","image":"730249423355.dkr.ecr.us-west-2.amazonaws.com/aws/aws-k8s-tester:latest"}
{"level":"info","ts":"2021-02-25T08:46:47.510Z","caller":"ecr/ecr.go:90","msg":"describing images","repo-name":"aws/aws-k8s-tester","repo-uri":"730249423355.dkr.ecr.us-west-2.amazonaws.com/aws/aws-k8s-tester","image-tag":"latest"}
{"level":"info","ts":"2021-02-25T08:46:47.525Z","caller":"ecr/ecr.go:111","msg":"described images","repo-name":"aws/aws-k8s-tester","image-tag":"latest","images":1}
{"level":"info","ts":"2021-02-25T08:46:47.525Z","caller":"ecr/ecr.go:117","msg":"found an image","index":0,"requested-tag":"latest","returned-tags":["latest"],"digest":"sha256:6b8a0b877bda31756deaf663d5ee3c4448998f8132b6aa2cee3ce77cde58f0ff","pushed-at":"2021-02-25 03:4
3:38 +0000 UTC","size":"319 MB"}
{"level":"info","ts":"2021-02-25T08:46:47.525Z","caller":"k8s-client/k8s.go:209","msg":"creating namespace","namespace":"eks-2021022507-spheres8fvv3-stresser-remote-v2"}
{"level":"info","ts":"2021-02-25T08:46:47.532Z","caller":"k8s-client/k8s.go:218","msg":"namespace already exists","namespace":"eks-2021022507-spheres8fvv3-stresser-remote-v2","error":"namespaces \"eks-2021022507-spheres8fvv3-stresser-remote-v2\" already exists"}
{"level":"info","ts":"2021-02-25T08:46:47.532Z","caller":"stresser2/stresser.go:199","msg":"creating stresser2 ServiceAccount"}
{"level":"info","ts":"2021-02-25T08:46:47.540Z","caller":"stresser2/stresser.go:226","msg":"created stresser2 ServiceAccount"}
{"level":"info","ts":"2021-02-25T08:46:47.552Z","caller":"stresser2/stresser.go:256","msg":"creating stresser RBAC ClusterRole"}
{"level":"info","ts":"2021-02-25T08:46:47.561Z","caller":"stresser2/stresser.go:315","msg":"created stresser2 RBAC ClusterRole"}
{"level":"info","ts":"2021-02-25T08:46:47.573Z","caller":"stresser2/stresser.go:347","msg":"creating stresser2 RBAC ClusterRoleBinding"}
{"level":"info","ts":"2021-02-25T08:46:47.580Z","caller":"stresser2/stresser.go:392","msg":"created stresser2 RBAC ClusterRoleBinding"}
{"level":"info","ts":"2021-02-25T08:46:47.593Z","caller":"stresser2/stresser.go:428","msg":"creating CronJob","name":"stresser2-app","completes":10,"parallels":10,"object-size":"1.4 kB"}
{"level":"info","ts":"2021-02-25T08:46:47.603Z","caller":"stresser2/stresser.go:444","msg":"created CronJob"}
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: eksconfig/add-on-stresser-remote-v2.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	// ObjectSize is the value size in bytes for write objects.
+	// If 0, do not write anything.
+	ObjectSize int `json:"object-size"`
+
+	// Duration is the duration to run stress2 testing.
+	Duration   time.Duration `json:"duration,omitempty"`
+	DurationString string    `json:"duration-string,omitempty" read-only:"true"`
+
+	// Coroutine is the number of concurrent go routines run per job
+	Coroutine  int `json:"coroutine"`
```
Comment (gyuho) [inline eksconfig/add-on-stresser-remote-v2.go:63]: `Coroutines`?

File: eksconfig/add-on-stresser-remote-v2.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	// SecretNum is the number of secrets generated per job
+	SecretNum  int `json:"secret-num"`
```
Comment (gyuho) [inline eksconfig/add-on-stresser-remote-v2.go:65]: `s/SecretNum/Secrets/`?

File: eksconfig/env_test.go [hunk 3]
```diff
@@ -584,6 +586,38 @@ spec:
...
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_PARALLELS")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SUCCESSFUL_JOBS_HISTORY_LIMIT", "500")
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SUCCESSFUL_JOBS_HISTORY_LIMIT")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_FAILED_JOBS_HISTORY_LIMIT", "500")
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_FAILED_JOBS_HISTORY_LIMIT")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_OBJECT_SIZE", "512")
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_OBJECT_SIZE")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_DURATION", "7m30s")
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_DURATION")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_COROUTINE", "10")
```
Comment (gyuho) [inline eksconfig/env_test.go:617]: `AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_COROUTINES`?

File: eksconfig/env_test.go [hunk 4]
```diff
@@ -584,6 +586,38 @@ spec:
...
+	defer os.Unsetenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_COROUTINE")
+	os.Setenv("AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SECRET_NUM", "10")
```
Comment (gyuho) [inline eksconfig/env_test.go:619]: `AWS_K8S_TESTER_EKS_ADD_ON_STRESSER_REMOTE_V2_SECRETS`?

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go [hunk 5]
```diff
@@ -0,0 +1,595 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-k8s-tester/pkg/randutil"
```
Comment (gyuho) [inline cmd/aws-k8s-tester/eks/create-stresser-v2.go:6]: Can we separate std package imports and 3rd party dependencies and run `./hack/fmt.sh`?

```
"context"
...

"github.com/..."
```

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+func newCreateStresserV2() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "stresser2",
+		Short: "Creates stresser v2",
+		Run:   createStresserFuncV2,
+	}
+	cmd.PersistentFlags().IntVar(&cfg.N, "number", 5, "number of go routines")
+	cmd.PersistentFlags().DurationVar(&cfg.duration, "duration", 10*time.Minute, "duration of the simulation")
+	cmd.PersistentFlags().IntVar(&cfg.objectSize, "object-size", 8, "object size, by default 8 bytes")
+	cmd.PersistentFlags().IntVar(&cfg.secretNum, "secret-num", 10, "secret object in default namespace, no more than 500")
```
Comment (gyuho) [inline cmd/aws-k8s-tester/eks/create-stresser-v2.go:37]: `secrets`?

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go (hunk 5 above)
Comment (chaochn47) [reply cmd/aws-k8s-tester/eks/create-stresser-v2.go:6]: yep

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go (hunk 6 above)
Comment (chaochn47) [reply cmd/aws-k8s-tester/eks/create-stresser-v2.go]: yep, updated.

File: eksconfig/env_test.go (hunk 4 above)
Comment (chaochn47) [reply eksconfig/env_test.go]: yep, updated.

File: eksconfig/env_test.go (hunk 3 above)
Comment (chaochn47) [reply eksconfig/env_test.go]: yep, updated.

File: eksconfig/add-on-stresser-remote-v2.go (hunk 2 above)
Comment (chaochn47) [reply eksconfig/add-on-stresser-remote-v2.go]: yep, updated.

File: eksconfig/add-on-stresser-remote-v2.go (hunk 1 above)
Comment (chaochn47) [reply eksconfig/add-on-stresser-remote-v2.go]: yep, updated.

**Review by gyuho — APPROVED** [review]

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go [hunk 7]
```diff
@@ -0,0 +1,595 @@
...
+							fmt.Println(fmt.Sprintf("time out stop updating the secret %s", secretName))
+							cancel()
+							return
+						case <-stopc:
+							fmt.Println("received stop signal")
+							return
+						default:
+						}
+						// Update Deployment
+						fmt.Printf("Updating secret %q.\n", secretName)
```
Comment (jqmichael) [inline cmd/aws-k8s-tester/eks/create-stresser-v2.go:182]: Update Secret?

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go [hunk 8]
```diff
@@ -0,0 +1,595 @@
...
+						err := cmClient.Delete(context.TODO(), cmName, metav1.DeleteOptions{})
+						if err != nil {
+							fmt.Printf("fail delete configmap %s due to %v.\n", cmName, err)
+						} else {
+							fmt.Printf("Deleted configmap %s.\n", cmName)
+						}
+					}()
+
+					// define the deployment object
+					cm := &corev1.ConfigMap{
```
Comment (jqmichael) [inline cmd/aws-k8s-tester/eks/create-stresser-v2.go:290]: Comment is off.

**Review by jqmichael — COMMENTED** [review]
Should we update README or provide instructions?

File: cmd/aws-k8s-tester/eks/create-stresser-v2.go (hunk 8 above)
Comment (chaochn47) [reply cmd/aws-k8s-tester/eks/create-stresser-v2.go:290]: yep, will update in next PR.

**Pull request merged**